### PR TITLE
dme - DNSMadeEasy Provider

### PR DESCRIPTION
**Pull request by soniah**
Pull request for dme - DNSMadeEasy Provider. See also issue #829.

The DNSMadeEasy REST API has probably grown over time, and is slightly
inconsistent. Most DNS record types take a 'value' field, however
this value field is sometimes named something else in the web interface.
Also, some record types require fields in addtion to 'value'.

I've attempted to document these requirements in my documentation - let me
know if you'd like me to lay out the documentation in a different manner, to
conform with your standards.

### CODE REVIEW

**Comment by soniah** [comment]
I removed my .travis.yml from my code - should I add it to the main .travis.yml?

Also, I ran my tests do the following, and they passed

```
TF_ACC=1 DME_AKEY=deadbeaf-deadbeef DME_SKEY=deadbeaf-deadbeef DME_DOMAINID=123456 DME_USESANDBOX='true'  go test ./builtin/providers/dnsmadeeasy -v  -timeout 45m
```
However doing the following gave me errors about unset environment vars (provider_test.go:34: DME_SKEY must be set for acceptance tests), I don't know how to setup testacc correctly:

```
make testacc TEST=./builtin/providers/dnsmadeeasy TESTARGS='DME_AKEY=deadbeef DME_SKEY=deadbeef DME_DOMAINID=123456 DME_USESANDBOX="true"'
```

**Comment by svanharmelen** [comment]
Something like this should work:

```
DME_AKEY=deadbeef DME_SKEY=deadbeef DME_DOMAINID=123456 DME_USESANDBOX="true" make testacc TEST=./builtin/providers/dnsmadeeasy
```

`TESTARGS` in this case is used to specify which test you want to run. If not specified it will run all tests.... See some additional info about `testacc` at the bottom of the [README.md](https://github.com/hashicorp/terraform/blob/master/README.md):

```
The TEST variable is required, and you should specify the folder where the provider is. The TESTARGS variable is recommended to filter down to a specific resource to test, since testing all of them at once can take a very long time.
```

**Comment by mitchellh** [comment]
This looks great! We're currently going through and doing a merge for 0.3.7, and we'll look at this for 0.4.0 since it is a new provider.

**Comment by hh** [comment]
TLDR; How do you create the terraform-provider-dme binary?

I'm new to go, and I couldn't find any documentation on how to actually build the binary for use with 0.3.6.

From https://github.com/soniah/terraform-provider-dme#installation

```
install the Terraform development environment, build it using the Development Environment instructions
```

I did a make dev and got no errors.

```
install this project via go get github.com/soniah/terraform-provider-dme
```

This doesn't produce any files on disk that I can tell, nor a binary.

```
follow the instructions for Installing a Plugin detailed in Plugin Basics
```

This seems to document installing a binary, which I can't seem to generate or find yet.

**Comment by soniah** [comment]
Hi Hippy @hh. Good call, I'll update the docs to make things more explicit -- give me a day or two -- $PHB is keeping me busy at $WORK at the moment.

**Comment by soniah** [comment]
Hi @hh, I've updated the README, I hope it helps.

**Comment by pearkes** [comment]
Thanks, this looks great. One thing is the bin is in the wrong place, but I'll fix that after merge and cc you on the commit.

I couldn't run the acceptance tests due to not having a small business account (paid) with DNSMadeEasy, but I assume those are passing as of the PR. If you can confirm that, it'd be appreciated.

Also, if you know of a way to set-up and account for testing for us to use, I'd appreciate it. Going to merge this now though.

**Comment by soniah** [comment]
Jack @pearkes quick reply to this and other stuff as I see you're working on this but I'm at work at the moment (Sydney time):

* bin fix -- thanks, great
* to setup a test account, you can get a $free 'sandbox' account from dnsmadeeasy. I then encrypted the keys and added them to travis. For example, see https://github.com/soniah/terraform-provider-dme/blob/master/.travis.yml
* "rename dnsmadeeasy package to dme" -- good idea

**Comment by hh** [comment]
Looks like we got a merge

**Comment by pearkes** [comment]
@soniah Cool, thanks, I'll sign up for that. Note with the rename that the binary named changed, so anything local you're running now will need to be updated if you're using master to build.
